<?xml version="1.0"?>
<launch>
  <node name="line_extractor" pkg="laser_line_extraction" type="line_extraction_node">
    <param name="~frequency" value="50.0" />
    <param name="~frame_id" value="laser_frame" />
    <param name="~scan_topic" value="scan" />
    <param name="~publish_markers" value="true" />
    <param name="~bearing_std_dev" value="0.0015" /> <!--0.0015-->
    <param name="~range_std_dev" value="0.01" /> <!--0.01-->
    <param name="~least_sq_angle_thresh" value="0.0001" /> <!--0.0001-->
    <param name="~least_sq_radius_thresh" value="0.0001" /> <!--0.0001-->
    <param name="~max_line_gap" value="0.5" /> <!--0.5 The maximum distance between two points in the same line (m).-->
    <param name="~min_line_length" value="0.03" /> <!--0.03 Lines shorter than this are not published (m).-->
    <param name="~min_range" value="0.1" />      <!--0.2 Points closer than this are ignored (m).-->
    <param name="~min_split_dist" value="0.05" />   <!--0.05 a split between two points results when the two points are at least this far apart (m)-->
    <param name="~outlier_dist" value="0.05" />  <!--0.05-->
    <param name="~min_line_points" value="5" /> <!--0.05-->
  </node>

  <node pkg="auto_dock" type="pattern" name="pattern_node" output="screen">
    <rosparam file="$(find param)/param_list.yaml" command="load"/>" 
  </node>
<!--
  <node pkg="auto_dock" type="controller" name="controller_node" output="screen">
  <param name="base_frame_id" value="base_link" />
  <param name="min_v" value="0.1" />
  <param name="min_w" value="0.1" />
  <param name="max_v" value="0.3" />
  <param name="max_w" value="0.3" />
  <param name="threshold_v" value="0.3" />
  <param name="threshold_w" value="0.4" />
  <param name="dist_to_dock" value="0.25" />
  <param name="dist_to_center" value="0.03" />
  </node> -->
</launch>
